src/transformers/tokenization_utils.py

```diff
@@ -770,3 +772,172 @@ def _decode(
             return clean_text
         else:
             return text
+
+
+class PreTrainedSentencepieceTokenizer(PreTrainedTokenizer):
```

Comment:
I'm not too keen on having an additional abstraction layer `PreTrainedSentencepieceTokenizer`. 

I thought the original idea was to replace instances of `"".join(tokens).replace(SPIECE_UNDERLINE, " ").strip() ` by `self._tokenizer.decode(tokens) `.

Could you explain why the changes proposed here are necessary?

Reply:
> Could you explain why the changes proposed here are necessary?

Well. While doing this refactoring I noticed a lot of duplicate code in the tokenizers implementations.
Since wet code is hard to maintain I tried to refactor it.

But if you do not like my refactoring I can just do the single change and that's it.